build/Tests/3rdparty/libDispatch/libDispatch.3rdPartyUT.vcxproj

```diff
@@ -221,34 +221,34 @@
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_drift.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_group.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_pingpong.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_priority.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_sema.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_starfish.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_test.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_timer_bit31.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_timer_bit63.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_timer_set_time.m" />
-    <ClangCompile Include="..\..\..\..\tests\unittests\libDispatch\dispatch_autorelease.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_after.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_api.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_apply.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_c99.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_cascade.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_debug.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_drift.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_group.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_pingpong.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_priority.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_sema.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_starfish.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_test.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_timer_bit31.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_timer_bit63.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_timer_set_time.m" />
+    <ClangCompile Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_autorelease.m" />
   </ItemGroup>
   <ItemGroup>
-    <ClInclude Include="..\..\..\..\tests\unittests\libDispatch\dispatch_test.h" />
+    <ClInclude Include="$(StarboardBasePath)\tests\3rdparty\libDispatch\dispatch_test.h" />
```

Comment:
We have taken hard dependency with Foundation and that is something I have talked with Meredith. For now these tests need to build from here unless we want to pull all WinObjC dependency into 3rd party folder (which I am not in favor of)

---
In reply to: [74129569](https://github.com/Microsoft/WinObjC/pull/853#discussion_r74129569) [](ancestors = 74129569)